src/jit/lowerxarch.cpp

```diff
@@ -1887,6 +1885,28 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTree* tree)
     TreeNodeInfo* info = &(tree->gtLsraInfo);
     LinearScan*   l    = m_lsra;
 
+#ifdef _TARGET_X86_
+    if (tree->gtOp.gtOp1->gtOper == GT_FIELD_LIST)
+    {
+        GenTreeFieldList* fieldListPtr   = tree->gtOp.gtOp1->AsFieldList();
+        for (; fieldListPtr; fieldListPtr = fieldListPtr->Rest())
```

Comment:
One of these days we should add a range-based-for-compatible iterator to GenTree lists.